activerecord/lib/active_record/connection_adapters/sqlite3/schema_definitions.rb

```diff
@@ -0,0 +1,23 @@
+module ActiveRecord
+  module ConnectionAdapters
+    module SQLite3
+      module ColumnMethods
+        def primary_key(name, type = :primary_key, **options)
+          if options.delete(:auto_increment) == true
+            type = :primary_key
```

Comment:
I intended to be passed `type = :integer, auto_increment = true` (implicit default primary key type) from the compatibility layer and then behaves as `:primary_key`. Otherwise we should keep the previous behavior. So I'll add `type == :integer` condition.